dspy/adapters/base.py

```diff
@@ -28,25 +32,37 @@ def __call__(
         demos: list[dict[str, Any]],
         inputs: dict[str, Any],
     ) -> list[dict[str, Any]]:
-        inputs = self.format(signature, demos, inputs)
-
-        outputs = lm(messages=inputs, **lm_kwargs)
-        values = []
-
-        for output in outputs:
-            output_logprobs = None
-
-            if isinstance(output, dict):
-                output, output_logprobs = output["text"], output["logprobs"]
-
-            value = self.parse(signature, output)
-
-            if output_logprobs is not None:
-                value["logprobs"] = output_logprobs
-
-            values.append(value)
-
-        return values
+        retry_count = 0
+        max_retries = max(settings.adapter_retry_count, 0)
+        outputs = None
+        while (True):
```

Comment:
nonblocking: I prefer to use a for-loop `for attempt in range(max_retries + 1)` to indicate that this is a loop with cap. Feel free to dismiss if you like the `while` structure better.